bottlecap/src/lifecycle/invocation/span_inferrer.rs

```diff
@@ -76,6 +76,8 @@ impl SpanInferrer {
             ..Default::default()
         };
 
+        let mut is_step_function = false;
```

Comment:
Can't we just downcast into the StepFunctions event instead of having a boolean? That way at the very bottom you can just check if the trigger is StepFunction

Reply:
Not sure what you mean

Reply:
instead of having a boolean, where you have this thing, just try downcasting the boxed trigger into a stepfunctions event, if it succeeds, then its guaranteed to be a stepfunction event

Reply:
can you give me an example? I don't think it's doable or considered a good approach in rust